src/domain/ledger/index.types.d.ts

```diff
@@ -0,0 +1,45 @@
+type LedgerError = import("./errors").LedgerError
+
+declare const liabilitiesAccountId: unique symbol
+type LiabilitiesAccountId = string & { [liabilitiesAccountId]: never }
+
+declare const ledgerTransactionIdSymbol: unique symbol
+type LedgerTransactionId = string & { [ledgerTransactionIdSymbol]: never }
+
+declare const ledgerAccountIdSymbol: unique symbol
+type LedgerAccountId = string & { [ledgerAccountIdSymbol]: never }
+
+type LedgerTransactionType =
+  typeof import("./index").LedgerTransactionType[keyof typeof import("./index").LedgerTransactionType]
```

Comment:
I believe we can type this directly here to get the constant values assigned to each field. Typescript should allow something like `type a = "Alice"` to be type string restricted to single string value `'Alice'`.

```suggestion
type LedgerTransactionType = {
  Invoice: "invoice"
  Payment: "payment"
  IntraLedger: "on_us"
  LnFeeReimbursement: "fee_reimbursement" // lightning
  OnchainReceipt: "onchain_receipt"
  OnchainPayment: "onchain_payment"
  OnchainIntraLedger: "onchain_on_us"
  OnchainDepositFee: "deposit_fee" // onchain
  Fee: "fee"
  Escrow: "escrow"
  RoutingFee: "routing_fee" // channel-related
  ExchangeRebalance: "exchange_rebalance" // send/receive btc from the exchange
  UserRebalance: "user_rebalance" // buy/sell btc in the user wallet
  ToColdStorage: "to_cold_storage"
  ToHotWallet: "to_hot_wallet"
}
```